text/src/tr.rs

```diff
@@ -148,78 +150,159 @@ fn create_minimal_string(chars: Vec<Char>, size: usize) -> Vec<char> {
     result
 }
 
+/// Parses a sequence in the format `[=equiv=]` from the given character iterator.
+///
+/// The function expects the iterator to be positioned just before the first `=`
+/// character. It reads the equivalent characters between the `=` symbols and
+/// creates a list of `Operand::Equiv` entries, one for each character.
+///
+/// # Arguments
+///
+/// * `chars` - A mutable reference to a peekable character iterator.
+///
+/// # Returns
+///
+/// A `Result` containing a vector of `Operand::Equiv` entries if successful, or a
+/// `String` describing the error if parsing fails.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// - The sequence does not contain a closing `=` before `]`.
+/// - The sequence does not contain a closing `]`.
+/// - The sequence contains no characters between the `=` symbols.
+///
+fn parse_equiv(chars: &mut std::iter::Peekable<std::str::Chars>) -> Result<Vec<Operand>, String> {
+    chars.next(); // Skip '='
+    let mut equiv = String::new();
+
+    while let Some(&next_ch) = chars.peek() {
+        if next_ch == '=' {
+            break;
+        }
+        equiv.push(next_ch);
+        chars.next();
+    }
+
+    if equiv.is_empty() {
+        return Err("Error: Missing equiv symbol after '[='".to_string());
+    }
+
+    let Some(&'=') = chars.peek() else {
+        return Err("Error: Missing '=' before ']' for '[=equiv=]'".to_string());
+    };
+    chars.next(); // Skip '='
```

Comment:
peek + skip = next in most cases 
```suggestion
    let Some(&'=') = chars.next() else {
        return Err("Error: Missing '=' before ']' for '[=equiv=]'".to_string());
    };
```